server/data/contacts.ts

```diff
@@ -0,0 +1,314 @@
+
+    const cycleId = user.cycles[user.cycles.length - 1];
+
+    const cyclesCollection: any = await cycles();
+    const cycle = await cyclesCollection.findOne({_id: new ObjectId(cycleId)});
+    if(cycle === null) throw new Error('Could not find cycle');
+
+    let application = null;
+    for(let element of cycle.applications){
+        if(element._id.toString() === applicationId){
+            application = element;
+            break;
+        }
+    }
+    if(application === null) throw new Error('Could not find application');
+    
+    let contacts = application.contacts ? application.contacts : [];
+
+    return contacts
+}
+
+/**
+ * Create contact
+ * @param {string} userId 
+ * @param {string} applicationId
+ * @param {string} title 
+ * @param {Date} date 
+ * @param {String} location 
+ * @returns {Promise<object>} if the contact was created, throws otherwise 
+ */
+ export const createContact = async (userId: string, applicationId: string, name: string, pronouns: string, location: string, phone: string, email: string): Promise<object> => {
+
+    if(checkObjectId(userId)){
+        throw new Error('Invalid userId');
+    }
+    if(checkObjectId(applicationId)){
+        throw new Error('Invalid applicationId');
+    }
+    if(checkName(name)){
+        throw new Error('Invalid name');
+    }
+    if(checkNonEmptyString(pronouns)){
+        throw new Error('Invalid pronouns');
+    }
+    if(checkNonEmptyString(location)){
+        throw new Error('Invalid location');
+    }
+    if(checkNonEmptyString(phone)){
+        throw new Error('Invalid phone number');
+    }
+    if(checkEmail(email)){
+        throw new Error('Invalid Email');
+    }
+
+    const usersCollection: any = await users();
+    const user = await usersCollection.findOne({_id: new ObjectId(userId)});
+    if(user === null) throw new Error('Could not find user');
+
+    const cycleId = user.cycles[user.cycles.length - 1];
+
+    const cyclesCollection: any = await cycles();
+    const cycle = await cyclesCollection.findOne({_id: new ObjectId(cycleId)});
+    if(cycle === null) throw new Error('Could not find cycle');
+
+    const newContact: ContactObject = {
+        _id: new ObjectId(),
+        name: name,
+        pronouns: pronouns,
+        location: location,
+        phone: phone,
+        email: email
+    }
+
+    let application = null;
+    for(let element of cycle.applications){
+        if(element._id.toString() === applicationId){
+            application = 1;
+            break;
+        } 
+    }
+    if(application === null) throw new Error('Could not find application');
+    
+    let updateInfo = await cyclesCollection.update(
+        {
+        _id: new ObjectId(cycleId),
+        "applications._id": new ObjectId(applicationId)
+        }, 
+        {$push: {
+            "applications.$.contacts": newContact
+            }
+        }
+    );
+    if (updateInfo.WriteResult.nModified === 0) throw new Error('Could not add contact');
```

Comment:
is this the correct syntax for checking if modified? if you tested it and it works, please resolve this comment